pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -253,10 +358,47 @@ func (service *SubnetPortService) DeleteSubnetPort(nsxSubnetPort *model.VpcSubne
 		return err
 	}
 	log.Info("successfully deleted nsxSubnetPort", "nsxSubnetPortID", *nsxSubnetPort.Id)
+	// delete DHCP static binding if needed
+	dhcpStaticBinding := service.DHCPStaticBindingStore.GetByKey(*nsxSubnetPort.Id)
+	if dhcpStaticBinding == nil || dhcpStaticBinding.Id == nil {
+		log.V(2).Info("DHCP static binding is not found in store, skip deleting it")
+	} else {
+		err := service.DeleteDHCPStaticBinding(subnetPortInfo, *dhcpStaticBinding.Id)
+		if err != nil {
+			log.Error(err, "Failed to delete DHCP static binding", "dhcpStaticBinding.Id", dhcpStaticBinding.Id)
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteDHCPStaticBinding deletes DHCP static binding.
+func (service *SubnetPortService) DeleteDHCPStaticBinding(subnetPortInfo servicecommon.VPCResourceInfo, bindingId string) error {
+	staticBinding := service.DHCPStaticBindingStore.GetByKey(bindingId)
+	if staticBinding != nil {
+		err := service.NSXClient.DhcpStaticBindingConfigsClient.Delete(subnetPortInfo.OrgID, subnetPortInfo.ProjectID, subnetPortInfo.VPCID, subnetPortInfo.ParentID, bindingId)
+		err = nsxutil.TransNSXApiError(err)
+		if err != nil {
+			log.Error(err, "Failed to delete DHCP static binding", "dhcpStaticBinding.Id", bindingId)
+			return err
+		}
+		service.DHCPStaticBindingStore.Delete(staticBinding)
+	}
 	return nil
 }
 
 func (service *SubnetPortService) DeleteSubnetPortById(portID string) error {
+	dhcpStaticBinding := service.DHCPStaticBindingStore.GetByKey(portID)
+	if dhcpStaticBinding == nil || dhcpStaticBinding.Id == nil {
+		log.V(2).Info("DHCP static binding is not found in store, skip deleting it")
+	} else {
+		dhcpStaticBindingInfo, _ := servicecommon.ParseVPCResourcePath(*dhcpStaticBinding.Path)
+		err := service.DeleteDHCPStaticBinding(dhcpStaticBindingInfo, *dhcpStaticBinding.Id)
+		if err != nil {
+			log.Error(err, "Failed to delete DHCP static binding", "dhcpStaticBinding.Id", dhcpStaticBinding.Id)
+			return err
+		}
+	}
```

Comment:
I saw in `DeleteSubnetPort` you delete subnetport before the static binding deletion. If the NSX operator crashed after the port deletion but before the binding deletion, it looks like that the GC cannot collect the binding ID?

Reply:
Good catch, I changed to delete static binding first then delete subnet port.